CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinkInventoryBase.h

```diff
@@ -0,0 +1,233 @@
+/*************************************************************************
+*    CompuCell - A software framework for multimodel simulations of     *
+* biocomplexity problems Copyright (C) 2003 University of Notre Dame,   *
+*                             Indiana                                   *
+*                                                                       *
+* This program is free software; IF YOU AGREE TO CITE USE OF CompuCell  *
+*  IN ALL RELATED RESEARCH PUBLICATIONS according to the terms of the   *
+*  CompuCell GNU General Public License RIDER you can redistribute it   *
+* and/or modify it under the terms of the GNU General Public License as *
+*  published by the Free Software Foundation; either version 2 of the   *
+*         License, or (at your option) any later version.               *
+*                                                                       *
+* This program is distributed in the hope that it will be useful, but   *
+*      WITHOUT ANY WARRANTY; without even the implied warranty of       *
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU    *
+*             General Public License for more details.                  *
+*                                                                       *
+*  You should have received a copy of the GNU General Public License    *
+*     along with this program; if not, write to the Free Software       *
+*      Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.        *
+*************************************************************************/
+#ifndef FOCALPOINTPLASTICITYLINKINVENTORYBASE_H
+#define FOCALPOINTPLASTICITYLINKINVENTORYBASE_H
+
+#include <map>
+#include <set>
+#include <vector>
+
+#include <CompuCell3D/CC3D.h>
+
+#include "FocalPointPlasticityLinks.h"
+#include "FocalPointPlasticityDLLSpecifier.h"
+
+namespace CompuCell3D {
+
+	/**
+	Written by T.J. Sego, Ph.D.
+	*/
+
+	class Potts3D;
+	class FocalPointPlasticityTracker;
+
+	typedef std::set<FocalPointPlasticityTrackerData> FPPTrackerDataSet;
+
+	template <typename T>
+	class FPPLinkListBase : public std::vector<T*> {
+	public:
+		typedef typename std::vector<T*>::iterator FPPLinkListIterator_t;
+		virtual ~FPPLinkListBase() {}
+	};
+
+	class FOCALPOINTPLASTICITY_EXPORT FPPLinkID {
+	public:
+		long id0;
+		long id1;
```

Comment:
if we go with std::unordered_map you will need to create hashing function too

Reply:
Implemented using Cantor pairing function in https://github.com/CompuCell3D/CompuCell3D/pull/361/commits/26d759b42d4cf99df801430ad6aeee84928218a4